DataCapturing/Persistence/BinarySerializer.swift

```diff
@@ -0,0 +1,299 @@
+    typealias Serializable = [AccelerationPointMO]
+
+    /**
+     Serializes an array of accelerations into binary format of the form:
+     - 8 Bytes: timestamp as long
+     - 8 Bytes: x as double
+     - 8 Bytes: y as double
+     - 8 Bytes: z as double
+
+     - Parameter accelerations: The array of accelerations to serialize.
+     - Returns: An array of serialized bytes.
+     */
+    func serialize(serializable accelerations: [AccelerationPointMO]) -> Data{
+        var ret = [UInt8]()
+        let byteOrder = ByteOrder.bigEndian
+
+        for acceleration in accelerations {
+            // 8 Bytes
+            ret.append(contentsOf: byteOrder.convertToBytes(acceleration.timestamp))
+            // 8 Bytes
+            ret.append(contentsOf: byteOrder.convertToBytes(acceleration.ax.bitPattern))
+            // 8 Bytes
+            ret.append(contentsOf: byteOrder.convertToBytes(acceleration.ay.bitPattern))
+            // 8 Bytes
+            ret.append(contentsOf: byteOrder.convertToBytes(acceleration.az.bitPattern))
+            // 32 Bytes
+        }
+
+        return Data(bytes: ret)
+    }
+}
+
+/**
+ A serializer for geo locations into the Cyface binary format representation
+
+ - Author: Klemens Muthmann
+ - Since: 2.0.0
+ - Version: 1.0.0
+ */
+class GeoLocationSerializer: BinarySerializer {
+    /// Binds the `Serializeble from the `BinarySerializer` protocoll to an array of geo locations.
+    typealias Serializable = [GeoLocationMO]
+
+    /**
+     Serializes an array of geo locations into binary format of the form:
+     - 8 Bytes: timestamp as long
+     - 8 Bytes: latitude as double
+     - 8 Bytes: longitude as double
+     - 8 Bytes: speed as double
+     - 4 Bytes: accuracy as int
+
+     - Parameter geoLocations: The array of locations to serialize.
+     - Returns: An array of serialized bytes.
+     */
+    func serialize(serializable locations: [GeoLocationMO]) -> Data {
+        var ret = [UInt8]()
+        let byteOrder = ByteOrder.bigEndian
+
+        for location in locations {
+            // 8 Bytes
+            let timestamp = location.timestamp
+            ret.append(contentsOf: byteOrder.convertToBytes(timestamp))
+            // 8 Bytes
+            let latBitPattern = location.lat.bitPattern
+            ret.append(contentsOf: byteOrder.convertToBytes(latBitPattern))
+            // 8 Bytes
+            let lonBitPattern = location.lon.bitPattern
+            ret.append(contentsOf: byteOrder.convertToBytes(lonBitPattern))
+            // 8 Bytes
+            let speedBitPattern = location.speed.bitPattern
+            ret.append(contentsOf: byteOrder.convertToBytes(speedBitPattern))
+            // 4 Bytes
+            let accuracy = UInt32(location.accuracy*100)
+            ret.append(contentsOf: byteOrder.convertToBytes(accuracy))
+            // = 36 Bytes
+        }
+
+        return Data(bytes: ret)
+    }
+}
+
```

Comment:
The next classes is a wrapper that produces a complete BinaryFormat Serializer from the output of the previous three classes.